ui/chapters/src/main/java/de/danoeh/antennapod/ui/chapters/PodcastIndexTranscriptUtils.java

```diff
@@ -37,6 +49,48 @@ public static String loadTranscriptFromUrl(String type, String url, boolean forc
         return str.toString();
     }
 
+    public static Transcript loadTranscript(FeedMedia media) {
+        String transcriptType = media.getItem().getPodcastIndexTranscriptType();
+
+        if (media.getItem().getPodcastIndexTranscriptText() != null) {
+            return TranscriptParser.parse(media.getItem().getPodcastIndexTranscriptText(), transcriptType);
+        }
+
+        if (media.getTranscriptFileUrl() != null) {
+            File transcriptFile = new File(media.getTranscriptFileUrl());
+
+            try {
+                if (transcriptFile != null && transcriptFile.exists()) {
+                    String t = FileUtils.readFileToString(transcriptFile, (String) null);
+                    if (StringUtils.isNotEmpty(t)) {
+                        return TranscriptParser.parse(t, transcriptType);
+                    }
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        String transcriptUrl = media.getItem().getPodcastIndexTranscriptUrl();
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        Future<Transcript> future = executor.submit(() -> {
+            String t = PodcastIndexTranscriptUtils.loadTranscriptFromUrl(transcriptType, transcriptUrl, false);
+            if (StringUtils.isNotEmpty(t)) {
+                return TranscriptParser.parse(t, transcriptType);
+            }
+            return null;
+        });
+
+        try {
+            Transcript result = future.get();  // This will block until the Callable completes
+            executor.shutdown();  // Remember to shutdown the executor
+            return result;
+        } catch (InterruptedException | ExecutionException e) {
+            e.printStackTrace();
+        }
```

Comment:
This looks like a workaround for `NetworkOnMainThreadException`. While it hides the exception, it still blocks the main thread until the download is done. Instead, the transcript should be loaded in the background without blocking the main thread. See `ChaptersFragment.loadMediaInfo`